src/titiler/core/titiler/core/algorithm/dem.py

```diff
@@ -224,9 +236,13 @@ def _range_check(datarange):
         if _range_check(datarange):
             raise ValueError(f"Data of {datarange} larger than 256 ** 3")
 
-        r = ((((data // 256) // 256) / 256) - (((data // 256) // 256) // 256)) * 256
-        g = (((data // 256) / 256) - ((data // 256) // 256)) * 256
-        b = ((data / 256) - (data // 256)) * 256
+        if self.use_nodata_height: 
+            data[img.array.mask[0]] = (self.nodata_height - self.baseval) / self.interval
+
+        data_int32 = data.astype(numpy.int32)
+        b = (data_int32) & 0xff 
+        g = (data_int32 >> 8) & 0xff
+        r = (data_int32 >> 16) & 0xff
```

Comment:
is that equivalent? or it does something new?

Reply:
It is doing things differently, qgis could not interpret the terrain-rgb endpoint as-is previously, but this one does work great. Maybe have to do with floating-point precision errors with that large 256^3 scaling?